### PR TITLE
Plugin: NoPendingCount

### PR DESCRIPTION
**Pull request by aamiaa**
This plugin removes the ping count of incoming friend requests, message requests, and nitro offers.
Ping indicators are removed in the following places:

- Home button (count reduced)
- Friends tab button
- Pending tab button
- Message Requests button
- Taskbar icon (count reduced)

The visibility logic of the Message Requests tab button is preserved (i.e. still appears when there are message requests, and disappears otherwise).

### CODE REVIEW

File: src/plugins/noPendingCount.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    settings: settings,
+
+    // Functions used to determine the top left count indicator can be found in the single module that calls getUnacknowledgedOffers(...)
+    // or by searching for "showProgressBadge:"
+    patches: [
+        {
+            find: ".getPendingCount=",
+            predicate: () => settings.store.hideFriendRequestsCount,
+            replacement: {
+                match: /(\i)\.getPendingCount=function\(\){return\s\i};/,
```
Comment (Vendicated) [inline src/plugins/noPendingCount.ts:62]:
```suggestion
                match: /\.getPendingCount=function\(\)\{/,
```

you can make this patch simpler by not unnecessarily matching \i and the function body and also using a lookbehind. Now your replacement only has to be `.getPendingCount=function(){return 0;`

You could make it even simpler by using a lookbehind:
```suggestion
                match: /(?<=\.getPendingCount=function\(\)\{)/,
```
Now your replacement becomes `return 0;`

File: src/plugins/noPendingCount.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                replace: "$1.getMessageRequestsCount=function(){return 0};"
+            }
+        },
+        // This prevents the Message Requests tab from always hiding due to the previous patch (and is compatible with spam requests)
+        // In short, only the red badge is hidden. Button visibility behavior isn't changed.
+        {
+            find: ".getSpamChannelsCount(),",
+            predicate: () => settings.store.hideMessageRequestsCount,
+            replacement: {
+                match: /(,\i=\i\.getSpamChannelsCount\(\)),(\i)=\i\.getMessageRequestsCount\(\),(\i=\i>0\|\|\i>0;return\s*?\i}})/,
```
Comment (Vendicated) [inline src/plugins/noPendingCount.ts:80]:
```suggestion
                match: /(?<=getSpamChannelsCount\(\),\i=)\i\.getMessageRequestsCount\(\)/,
```

Now your replacement is just `$self.getRealMessageRequestCount()`

File: src/plugins/noPendingCount.ts (hunk 1 above)
Comment (aamiaa) [reply src/plugins/noPendingCount.ts:62]: ah okay, for some reason I thought that causing the function to have double `return` would be bad

Comment (Vendicated) [reply src/plugins/noPendingCount.ts:62]: no it's completely fine, we do it a lot as it makes many things a lot simpler

**Pull request merged**